pyomo/dae/init_cond.py

```diff
@@ -0,0 +1,174 @@
+#  Under the terms of Contract DE-NA0003525 with National Technology and 
+#  Engineering Solutions of Sandia, LLC, the U.S. Government retains certain 
+#  rights in this software.
+#  This software is distributed under the 3-clause BSD License.
+#  ___________________________________________________________________________
+
+from pyomo.environ import Constraint, Block, value
+from pyomo.kernel import ComponentSet, ComponentMap
+from pyomo.dae.set_utils import (is_explicitly_indexed_by, get_index_set_except,
+        is_in_block_indexed_by)
+
+
+def index_warning(name, index):
+    return 'WARNING: %s has no index %s' % (name, index)
+
+
+def deactivate_model_at(b, cset, pts, allow_skip=True, suppress_warnings=False):
+    """
+    Finds any block or constraint in block b, indexed explicitly (and not 
+    implicitly) by cset, and deactivates it at points specified. 
+    Implicitly indexed components are excluded because one of their parent 
+    blocks will be deactivated, so deactivating them too would be redundant.
+
+    Args:
+        b : Block to search
+        cset : ContinuousSet of interest
+        pts : Value or list of values, in ContinuousSet, to deactivate at
+
+    Returns:
+        A dictionary mapping points in pts to lists of
+        component data that have been deactivated there
+    """
+    if not type(pts) is list:
+        pts = [pts]
+    for pt in pts:
+        if not pt in cset:
+            msg = str(pt) + ' is not in ContinuousSet ' + cset.name
+            raise ValueError(msg)
+    deactivated = {pt: [] for pt in pts}
+
+    visited = set()
+    for comp in b.component_objects([Block, Constraint], active=True):
+        # Record components that have been visited in case component_objects
+        # contains duplicates (due to references)
+        if id(comp) in visited:
+            continue
+        visited.add(id(comp))
+
+        if (is_explicitly_indexed_by(comp, cset) and
+                not is_in_block_indexed_by(comp, cset)):
+            info = get_index_set_except(comp, cset)
+            non_cset_set = info['set_except']
+            index_getter = info['index_getter']
+
+            for non_cset_index in non_cset_set:
+                for pt in pts:
+                    index = index_getter(non_cset_index, pt)
+                    try:
+                        comp[index].deactivate()
+                        deactivated[pt].append(comp[index])
+                    except KeyError:
+                        # except KeyError to allow Constraint/Block.Skip
+                        if not suppress_warnings:
+                            print(index_warning(comp.name, index))
+                        if not allow_skip:
+                            raise
+                        continue
+
+    return deactivated
+
+
+def get_inconsistent_initial_conditions(model, time, tol=1e-8, t0=None, 
+        allow_skip=True, suppress_warnings=False):
```

Comment:
Something that I've struggled with in my own work is how to set default tolerances. Is there a justification for `1e-8`?

Reply:
It's very small, but not zero. About square root of machine precision? I have no idea